sdks/v4-sdk/src/entities/pool.ts

```diff
@@ -0,0 +1,259 @@
+import invariant from 'tiny-invariant'
+import { keccak256 } from '@ethersproject/solidity'
+import { BigintIsh, Currency, CurrencyAmount, Price } from '@uniswap/sdk-core'
+import {
+  v3Swap,
+  NoTickDataProvider,
+  Tick,
+  TickConstructorArgs,
+  TickDataProvider,
+  TickListDataProvider,
+  TickMath,
+} from '@uniswap/v3-sdk'
+import { defaultAbiCoder } from 'ethers/lib/utils'
+import { sortsBefore } from '../utils/sortsBefore'
+import { ADDRESS_ZERO, NEGATIVE_ONE, Q192 } from '../utils/internalConstants'
+import JSBI from 'jsbi'
+
+const NO_TICK_DATA_PROVIDER_DEFAULT = new NoTickDataProvider()
+
+/**
+ * Represents a V4 pool
+ */
+export class Pool {
+  public readonly currency0: Currency
+  public readonly currency1: Currency
+  public readonly fee: number
+  public readonly tickSpacing: number
+  public readonly sqrtRatioX96: JSBI
+  public readonly hooks: string // address
+  public readonly liquidity: JSBI
+  public readonly tickCurrent: number
+  public readonly tickDataProvider: TickDataProvider
+
+  private _currency0Price?: Price<Currency, Currency>
+  private _currency1Price?: Price<Currency, Currency>
+
+  public static getPoolId(
```

Comment:
we might get the benefit of consistency between client library and protocol contract, if we define a `PoolKey` type in v4-sdk. That way, getPoolId method parameters are also more flexible to add/remove new params without modifying the getPoolId function itself.